### PR TITLE
refactor custom app routing

### PR DESCRIPTION
**Pull request by snopoke**
##### SUMMARY
Refactor app custom app routing to be defined in settings

related localsettings change: https://github.com/dimagi/commcare-cloud/pull/3972

### CODE REVIEW

File: corehq/sql_db/routers.py [hunk 1]
```diff
@@ -125,13 +117,12 @@ def db_for_read_write(model, write=True, hints=None):
 
     if app_label == SYNCLOGS_APP:
         return settings.SYNCLOGS_SQL_DB_ALIAS
-    elif app_label == ICDS_REPORTS_APP:
-        return connection_manager.get_django_db_alias(ICDS_UCR_CITUS_ENGINE_ID)
-    elif app_label == AAA_APP:
-        engine_id = AAA_DB_ENGINE_ID
-        if not write:
-            return connection_manager.get_load_balanced_read_db_alias(AAA_DB_ENGINE_ID)
```
Comment (snopoke) [inline corehq/sql_db/routers.py:133]: this load balancing is removed but it's not active on any environment anyway. I think the original code was just copied from the ICDS code at the time.

**Comment by snopoke** [comment]
This is now love on `icds-staging` env

File: settings.py [hunk 2]
```diff
@@ -1396,6 +1398,13 @@
 if 'corehq.sql_db.routers.MultiDBRouter' not in DATABASE_ROUTERS:
     DATABASE_ROUTERS.append('corehq.sql_db.routers.MultiDBRouter')
 
+# Mapping of app_label to DB name or reporting DB alias (see REPORTING_DATABASES)
+CUSTOM_DB_ROUTING = {
```
Comment (sravfeyn) [inline settings.py:1402]: Isn't the plan to do this via commcare-cloud setting override?

Comment (snopoke) [reply settings.py:1402]: https://github.com/dimagi/commcare-cloud/pull/3972/files#diff-6604a8eb62ac7594d06f38182d65ffa2R374-R375

**Review by sravfeyn — APPROVED** [review]

**Pull request merged**